install/kubernetes/helm/istio/charts/sidecar-injector/templates/mutatingwebhook.yaml

```diff
@@ -1,22 +1,20 @@
-{{- if .Capabilities.APIVersions.Has "admissionregistration.k8s.io/v1beta1" -}}
```

Comment:
Doesn't this feature still need to be gated on admission registration API?

Reply:
I tested it locally It didn't work with my Helm. Also I think if the cluster version does not support webhook, the installation should fail instead of silently ignore it.

Reply:
@yusuoh the Capabilities feature does not function with `helm template` - only with `helm install`.  The rationale behind not failing is to permit manual injection in the event that automatic injection doesn't work.  This could be changed (to a flag in `values.yaml` representing if webhooks are enabled.  If they are enabled, and the cluster does not have the webhook admission controller configured, a failure in deploy would produce least surprise.  Unfortunately not sure how this could be activated in `helm template`.

Reply:
@jascott1 any ideas on this particular issue?

Reply:
💯 we need to make sure this works with `helm template` as sdake said. I worked around it in #4402 but a more thorough solution would be great.

Reply:
I think it's reasonable for 'heml template' to trust the user to enable the feature explicitly.
So maybe rewrite this to check if Capabilities exists, and if not just rely on the '.enabled'.